layers.go

```diff
@@ -523,7 +523,7 @@ func (s *store) newLayerStore(rundir string, layerdir string, driver drivers.Dri
 }
 
 func newROLayerStore(rundir string, layerdir string, driver drivers.Driver) (ROLayerStore, error) {
-	lockfile, err := GetROLockfile(filepath.Join(layerdir, "layers.lock"))
+	lockfile, err := GetROLockfile(filepath.Join(layerdir, "layers-ro.lock"))
```

Comment:
Since it's now using a different lock file, how does a read-only store detect that the contents that are protected by the lock have been modified by another process that presumably took a write lock?

Reply:
I see this might be a problem. What I now did in https://github.com/containers/storage/pull/473/commits/9941e40a94d55291e24edfd9ea472f1b4d6718ee is adding an additional check before actually creating the store. Do you think we have to add the same pre-check to the `newLayerStore` method as well?

Reply:
`newLayerStore` and `newROLayerStore` alway calls `Load()`, so they're fine.  A long-running process isn't necessarily ever going to refresh the read-only store's contents after that, which I think is going to be a problem.

Reply:
So do you think we can solve this by acquiring a `layers.lock` and `layers-ro.lock` together on read-write?

Reply:
Yes, acquiring both locks, in an order consistent with other cases where both can be acquired at the same time, should avoid this.  I think doing so brings back the lock contention that you're trying to avoid by using a different lock, though, and I'm not sure what to do about that.

Reply:
I think if we adapt the locker implementation then we could solve the problem. I now added a more generic lockfile creation function `GetLockfileRWRO()`, which is able to change the locker type (read-only or not) during its usage.

Let's see what the tests say.